test-ts/TrueRewardBackedToken.test.ts

```diff
@@ -643,5 +643,115 @@ describe('TrueRewardBackedToken', () => {
+
+        await expect(financialOpportunity.awardPool()).to.emit(financialOpportunity, 'awardPoolSuccess').withArgs('11453250670444388466')
+
+        expect(await token.balanceOf(mockPoolAddress)).to.equal('17179876005666582699')
+        expect(await financialOpportunity.getBalance()).to.equal('132820123994333417300')
+      })
+
+      it('awards 0 on subsequent calls', async () => {
+        await financialOpportunity.setRewardBasis(0.7 * 1000)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+
+        await financialOpportunity.awardPool()
+
+        expect(await financialOpportunity.awardAmount()).to.equal(0)
+        await financialOpportunity.awardPool()
+        expect(await token.balanceOf(mockPoolAddress)).to.equal('17179876005666582699')
+        expect(await financialOpportunity.getBalance()).to.equal('132820123994333417300')
+      })
+
+      it('awards proper amount when per token value increases between calls', async () => {
+        await financialOpportunity.setRewardBasis(0.7 * 1000)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+
+        const firstTUsdAmount = '11453250670444388466' // 100 * (1.5 - 1.5 ^ 0.7) / 1.5
+        const secondTUsdAmount = '12580970036318224093' // 100 * (2.5 - 2.5 ^ 0.7) / 2.5 - firstTUsdAmount
+
+        await expect(financialOpportunity.awardPool()).to.emit(financialOpportunity, 'awardPoolSuccess').withArgs(firstTUsdAmount)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('2500000000'))
+        await expect(financialOpportunity.awardPool()).to.emit(financialOpportunity, 'awardPoolSuccess').withArgs(secondTUsdAmount)
+
+        // 1 wei error
+        expect(await token.balanceOf(mockPoolAddress)).to.equal('48632301096462142932') // firstTUsdAmount * 1.5 + secondTUsdAmount * 2.5
+        expect(await aaveFinancialOpportunity.getBalance()).to.equal('189914448233093468600') // (100 - firstTUsdAmount - secondTUsdAmount) * 2.5
+      })
+
+      it('no additional awards when reward basis changes between calls', async () => {
+        await financialOpportunity.setRewardBasis(0.7 * 1000)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+
+        await financialOpportunity.awardPool()
+        await financialOpportunity.setRewardBasis(0.5 * 1000)
+
+        expect(await financialOpportunity.awardAmount()).to.equal(0)
+        await financialOpportunity.awardPool()
+
+        expect(await token.balanceOf(mockPoolAddress)).to.equal('17179876005666582699')
+        expect(await financialOpportunity.getBalance()).to.equal('132820123994333417300')
+      })
+
+      it('does NOT revert if the withdrawal fails', async () => {
+        await financialOpportunity.setRewardBasis(0.7 * 1000)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('20000000000')) // 20x
+
+        await financialOpportunity.awardPool()
```

Comment:
await expect(financialOpportunity.awardPool()).to.emit(financialOpportunity, 'awardPoolFailure').withArgs('0')